src/lib/get-participations.ts

```diff
@@ -0,0 +1,20 @@
+import type { Participations } from '@guardian/ab-core';
+import { isObject, isString, storage } from '@guardian/libs';
+
+const participationsKey = 'gu.ab._participations';
+
+const isParticipations = (
+	participations: unknown,
+): participations is Participations => {
+	if (!isObject(participations)) return false;
+	return Object.values(participations).some((participation) => {
+		if (!isObject(participation)) return false;
+		if (!isString(participation.variant)) return false;
+		return true;
+	});
```

Comment:
This may come down to taste, but I personally find the early returns add some noise here.

Could the same logic be expressed by checking that `every` participation matches the criteria?

```suggestion
	isObject(participations) &&
	Object.values(participations).every(
		(participation) =>
			isObject(participation) && isString(participation.variant),
	);
```

I'll leave this up to you though!

Reply:
Hard to know what the original intention was because this case [isn't tested in DCR where this code was lifted from](https://github.com/guardian/dotcom-rendering/blob/67b249e29c8c45e72f5869483cbd658c1cb300ff/dotcom-rendering/src/web/experiments/lib/ab-localstorage.test.ts)

Reply:
So [this is the type](https://github.com/guardian/ab-testing/blob/b5de0e09259d99399f192e4696470812b2aa45b1/packages/ab-core/src/types.ts#L99) that we're testing against I think, which looks like every variant value must be a string.

Reply:
You're right, I'll update to the one you suggested, thanks